plugins/builds/update.js

```diff
@@ -96,8 +104,9 @@ module.exports = () => ({
                         buildLink:
                             `${buildFactory.uiUri}/pipelines/${pipeline.id}/builds/${id}`
                     });
-                    // Guard against triggering non-successful builds
-                    if (desiredStatus !== 'SUCCESS') {
+
+                    // Guard against triggering non-successful or unstable builds
+                    if (desiredStatus !== 'SUCCESS' || build.status === 'UNSTABLE') {
```

Comment:
won't `if (build.status !== 'SUCCESS')` do ?

Reply:
`build.status` is current status. `desiredStatus` is the new status

We only want to trigger if the new status is `SUCCESS`, and the current status is not `UNSTABLE`.

Reply:
Oh I see at this point `build.status` should be the new status already. We can use that check instead like you said.